### PR TITLE
[Testing] Speed up execution of flaky test group and improve test coverage of test listeners

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

Since the improvements in tests have been made to reduce resource leaks, it is now possible to remove `-DtestReuseFork=false` parameter used in the execution of the BROKER_FLAKY test group (part of `CI - Unit - Brokers - Other` workflow).

This will make the execution of  a lot faster. The execution time drops from [1h 20mins-1h 40mins](https://github.com/apache/pulsar/runs/2362917547?check_suite_focus=true) to [about 45 minutes](https://github.com/lhotari/pulsar/runs/2364160310?check_suite_focus=true).

This PR also includes some minor improvements to test listeners and adds better test coverage to verify the behavior.

### Modifications

- remove `-DtestReuseFork=false` and change `-DtestForkCount=1` to `-DtestForkCount=2` for `CI - Unit - Brokers - Other`/`BROKER_FLAKY` test group.
- add better test coverage for TestNG listeners in buildtools
- decouple PulsarTestListener and FailFastNotifier

### CODE REVIEW

**Comment by lhotari** [comment]
@aahmed-se @merlimat @sijie @eolivelli Please review. This speeds up execution of `CI - Unit - Brokers - Other`/`BROKER_FLAKY` test group (reduces to about 45 mins).

File: buildtools/pom.xml [hunk 1]
```diff
@@ -44,6 +44,7 @@
     <testng.version>7.3.0</testng.version>
     <commons-lang3.version>3.11</commons-lang3.version>
     <maven-shade-plugin.version>3.2.4</maven-shade-plugin.version>
+    <excludedGroups>myExcludedGroup</excludedGroups>
```
Comment (aahmed-se) [inline buildtools/pom.xml:47]: what is this group about myExcludedGroup

Comment (lhotari) [reply buildtools/pom.xml:47]: it's just to test within buildtools tests that the exclusions work as expected.

Comment (lhotari) [reply buildtools/pom.xml:47]: I had some doubts so that's why I added it. I can remove it if it is unnecessary.

Comment (lhotari) [reply buildtools/pom.xml]: I removed it. :)

**Review by eolivelli — APPROVED** [review]
LGTM

**Review by aahmed-se — APPROVED** [review]

**Comment by lhotari** [comment]
[Integration test builds suddenly broke](https://github.com/apache/pulsar/pull/10249/checks?check_run_id=2364929547#step:12:343) since building the docker image tries to download golang from https://storage.googleapis.com/golang/go1.13.3.linux-amd64.tar.gz . The correct URL is https://golang.org/dl/go1.13.3.linux-amd64.tar.gz . I have put the fix as part of this PR. Please review and merge this PR to unblock CI @aahmed-se @merlimat @sijie

**Pull request merged**